contracts/contracts/rem.swap/src/rem.swap.cpp

```diff
@@ -0,0 +1,307 @@
+/**
+ *  @copyright defined in eos/LICENSE.txt
+ */
+
+#include <rem.swap/rem.swap.hpp>
+
+namespace eosio {
+
+    void swap::init( const name& user, const public_key& swap_key, const string& trx_id,
+                     const string& chain_id, const string eth_address, const asset& quantity,
+                     const uint32_t& swap_init_time ) {
+
+        require_auth( user );
+        swap_index swap_table( _self, get_first_receiver().value );
+
+        check( quantity.is_valid(), "invalid quantity" );
+        check( chain_id == remchain_id, "not correct chainID" );
+        check( quantity.symbol == create_account_fee.symbol, "symbol precision mismatch" );
+        check( quantity.amount > create_account_fee.amount, "the quantity must be greater "
+                                                            "than the account creation fee" );
+
+        const time_point swap_init_timepoint = time_point(seconds(swap_init_time));
+
+        std::vector<char> swap_payload( swap_key.data.begin(), swap_key.data.end() );
+        string payload_str = trx_id + chain_id + eth_address + quantity.to_string()
```

Comment:
swap_id (or swap_hash) should sha256(trx_id+chain_id+swap_pub_key+quantity+swap_init_time)
public key should be in payload, not eth_address.